stream/nats.go

```diff
@@ -76,3 +90,27 @@ func getNatsTLSFromConfig() ([]nats.Option, error) {
 
 	return opts, nil
 }
+
+func setupConnOptions() []nats.Option {
+	return []nats.Option{
+		nats.Name(cfg.Config.NodeName()),
+		nats.RetryOnFailedConnect(true),
+		nats.ReconnectWait(time.Second),
```

Comment:
Let's add option in configuration to customize this as well. So that people can configure wait time.